group_utils/README.md

```diff
@@ -0,0 +1,171 @@
+
+# Purpose
+Get the names, labels or states of the direct or all members of a group. 
+Perform the arithmetic operations known from the openHAB [group item definition](https://www.openhab.org/docs/configuration/items.html#derive-group-state-from-member-items) on the states or count how many members match a given comparison function.
+
+# How it works
+
+## JavaScript
+The JavaScript version of this library was created and tested on openHAB 3.0 and 3.1. 
+This library has no dependencies. 
+It __does not work__ with the GraalVM JavaScript add-on.
+
+### Import the group_utils
+```javascript
+var OPENHAB_CONF = Java.type("java.lang.System").getenv("OPENHAB_CONF")
+load(OPENHAB_CONF + '/automation/lib/javascript/community/groupUtils.js')
+```
+
+### Arguments for all functions
+Argument | Purpose | Required
+-|-|-
+`group` | The name of the group. | yes
+
+### __Get something from members__
+
+### getMembersNames
+This function returns the __names__ of direct group members as an array.
+```javascript
+var groupNames = getMembersNames(group)
+```
+
+### getMembersStates
+This function returns the __states__ of direct group members as an array.
+```javascript
+var groupStates = getMembersStates(group)
+```
+
+### getMembersLabels
+This function returns the __labels__ of direct group members as an array.
+```javascript
+var groupLabels = getMembersLabels(group)
+```
+
+### getMembersLabelsString
+This function returns the __labels__ of direct group members as a concatenated string.
+```javascript
+var groupLabelsString = getMembersLabelsString(group)
+```
+
+### getAllMembersNames
+This function returns the __names__ of direct and child group members as an array.
+The group items are excluded.
+```javascript
+var groupAllNames = getAllMembersNames(group)
+```
+
+### getAllMembersStates
+This function returns the __states__ of direct and child group members as an array.
+The group items are excluded.
+```javascript
+var groupAllStates = getAllMembersStates(group)
+```
+
+### getAllMembersLabels
+This function returns the __labels__ of direct and child group members as an array.
+The group items are excluded.
+```javascript
+var groupAllLabels = getAllMembersLabels(group)
+```
+
+### getAllMembersLabelsStrings
+This function returns the __labels__ of direct and child group members as a concatenated string.
+The group items are excluded.
+```javascript
+var groupAllLabelsString = getAllMembersLabelsString(group)
+```
+
+### __Arithmetic operations__
+Perform arithmetic operations on the states of Number members. 
+This functionality is the same as in the openHAB [group item definition](https://www.openhab.org/docs/configuration/items.html#derive-group-state-from-member-items).
+The following functions only use Items of type: Number, Dimmer or Rollershutter.
+Other Items and NULL or UNDEF states are ignored.
+
+### MembersSum
+Get the __SUM__ of direct members' number states.
```

Comment:
This will apply to all the math functions here. Describe what happens when there are members of the Group that do not support math (e.g. NULL and UNDEF Items).

Reply:
Okay, does it go when I write:
`The following functions only use Items of type: Number, Dimmer or Rollershutter.
Members of other types or with NULL or UNDEF states are ignored, therefore they are excluded from all math functions.`?